lib/Sema/CSDiag.cpp

```diff
@@ -2617,7 +2618,7 @@ diagnoseTypeMemberOnInstanceLookup(Type baseObjTy,
         // Fetch any declaration to check if the name is '~='
         ValueDecl *decl0 = overloadedFn->getDecls()[0];
 
-        if (decl0->getName() == decl0->getASTContext().Id_MatchOperator) {
+        if (decl0->getBaseName() == decl0->getASTContext().Id_MatchOperator) {
```

Comment:
Hm, okay. I guess it's questionable whether we'd make `self` a special name, even though we clearly want to do so for things like `init`.